### DISCUSSION TITLE
Weekday-based template sections

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [SilentVoid13/Templater#828] [Help] [answered]
Hello,

I have a couple of recurring tasks that I set up in my daily-notes based on the day of the week. Meaning if I have a reminder for a stream which launches on monday/Wednesday/Friday@2015, I always include this as a task in my dailynote on those days.

The problem is I don't have only one such task - I counted 17. 
And I _really_ don't feel like continuously putting these in manually, so I'd like to get some way of including arbitrary text strings based on weekday.

Is that possible? 

Thank you.

Sincerely,
~Gw

---
Functionally this is identical to #843

**Comment by AB1908** [comment]
Hmm I feel that the idea itself doesn't seem too arcane but you'd likely need some more JS for this. Which part are you stuck with?

**Reply by Gewerd-Strauss** [reply]
ah frick. 

I intended to close this one, it has been resolved two days ago,  but I've been way too preoccupied with other tasks the last couple days.

**Comment by Gewerd-Strauss** [comment, marked as answer]
After a bit of tinkering, I came up with the following:


Within the DailyNote-Template:    
    ### Start of Day (<% this.app.workspace.getActiveFile().basename %>)
    <%  tp.file.include("{{Morning - 0" + tp.date.now("d dddd",0,this.app.workspace.getActiveFile().basename,"DD.MM.YYYY") + "}}") %>
    ### The Day (<% this.app.workspace.getActiveFile().basename %>)
    <%  tp.file.include("{{Day - 0" + tp.date.now("d dddd",0,this.app.workspace.getActiveFile().basename,"DD.MM.YYYY") + "}}") %>
    ### Evening (<% this.app.workspace.getActiveFile().basename %>)
    <%  tp.file.include("{{Evening - 0" + tp.date.now("d dddd",0,this.app.workspace.getActiveFile().basename,"DD.MM.YYYY") + "}}") %>
    
the respective templates are called

    Day - 00 Sunday
    Evening  01 Monday
    Morning 06 Saturday
    and so on
    
`this.app.workspace.getActiveFile().basename` is used because I pre-create my dailynotes via various plugins, and thus cannot use the simpler `tp.date.now` as reference.

---
This also solves #843.
